src/Models/Tag.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace CachetHQ\Cachet\Models;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Str;
+use Watson\Validating\ValidatingTrait;
+
+class Tag extends Model
+{
+    use ValidatingTrait;
+
+    /**
+     * The fillable properties.
+     *
+     * @var string[]
+     */
+    protected $fillable = ['name'];
+
+    /**
+     * Overrides the models boot method.
+     *
+     * @return void
+     */
+    public static function boot()
+    {
+        parent::boot();
+
+        self::creating(function ($tag) {
+            $tag->slug = Str::slug($tag->name);
```

Comment:
The slug is stored in the DB. This is setting the value to it. If the API changes how a slug is made then we will write our own. A slug is a slug though. 

Sent from my iPhone

> On 16 Jan 2015, at 19:02, Graham Campbell <notifications@github.com> wrote:
> 
> In src/Models/Tag.php:
> 
> > +     *
> > +     * @var string[]
> > +     */
> > +    protected $fillable = ['name'];
> > +
> > +    /**
> > +     * Overrides the models boot method.
> > +     *
> > +     * @return void
> > +     */
> > +    public static function boot()
> > +    {
> > +        parent::boot();
> > +
> > +        self::creating(function ($tag) {
> > +            $tag->slug = Str::slug($tag->name);
> This is bad. Slugs should actually be stored in the db. The laravel "slug" function may actually change, so cannot be relied upon.
> 
> —
> Reply to this email directly or view it on GitHub.
>